invenio_rdm_records/marshmallow/json.py

```diff
@@ -12,53 +12,125 @@
+    condition = SanitizedUnicode()
+    link_validity = SanitizedUnicode()
 
 
-class PersonIdsSchemaV1(StrictKeysMixin):
-    """Ids schema."""
+class CommunitySchemaV1(BaseSchema):
+    """Communities to which the record belongs to."""
 
-    source = SanitizedUnicode()
-    value = SanitizedUnicode()
+    primary = SanitizedUnicode(required=True)
+    secondary = fields.List(SanitizedUnicode())
 
 
-class ContributorSchemaV1(StrictKeysMixin):
+class IdentifierSchemaV1(BaseSchema):
+    """Extra/Alternate identifiers of the record."""
+
+    identifier = SanitizedUnicode(required=True)
+    scheme = SanitizedUnicode(required=True)
+
+
+class AffiliationSchemaV1(BaseSchema):
+    """Affiliation of a creator/contributor."""
+
+    name = SanitizedUnicode(required=True)
+    identifier = SanitizedUnicode(required=True)
+    scheme = SanitizedUnicode(required=True)
+
+
+class CreatorSchemaV1(BaseSchema):
+    """Creator schema."""
+
+    NAMES = [
+        "Organizational",
+        "Personal"
+    ]
+
+    name = SanitizedUnicode(required=True)
+    type = SanitizedUnicode(required=True, validate=validate.OneOf(
+                choices=NAMES,
+                error=_('Invalid name type. {input} not one of {choices}.')
+            ))
+    given_name = SanitizedUnicode()
+    family_name = SanitizedUnicode()
+    identifiers = fields.List(fields.Nested(IdentifierSchemaV1))
+    affiliations = fields.List(fields.Nested(AffiliationSchemaV1))
+
+
+class ContributorSchemaV1(CreatorSchemaV1):
     """Contributor schema."""
 
     ROLES = [
         "ContactPerson",
+        "DataCollector",
+        "DataCurator",
+        "DataManager",
+        "Distributor",
+        "Editor",
+        "HostingInstitution",
+        "Producer",
+        "ProjectLeader",
+        "ProjectManager",
+        "ProjectMember",
+        "RegistrationAgency",
+        "RegistrationAuthority",
+        "RelatedPerson",
         "Researcher",
+        "ResearchGroup",
+        "RightsHolder",
+        "Sponsor",
+        "Supervisor",
+        "WorkPackageLeader",
         "Other"
     ]
 
-    ids = fields.Nested(PersonIdsSchemaV1, many=True)
-    name = SanitizedUnicode(required=True)
-    role = SanitizedUnicode(
-        validate=validate.OneOf(
-            choices=ROLES,
-            error=_('Invalid role. {input} not one of {choices}.')
-        ))
-    affiliations = fields.List(SanitizedUnicode())
-    email = fields.Email()
+    role = SanitizedUnicode(required=True, validate=validate.OneOf(
+                choices=ROLES,
+                error=_('Invalid role. {input} not one of {choices}.')
+            ))
+
+
+class FilesSchemaV1(BaseSchema):
+    """Files metadata schema."""
+
+    type = fields.String(required=True)
+    checksum = fields.String(required=True)
+    size = fields.Integer(required=True)
+    bucket = fields.String(required=True)
+    key = fields.String(required=True)
+    links = fields.Method('get_links', required=True)
```

Comment:
Why exactly? Being a Method it would be there before validation and therefore checking it is there, right?

Reply:
`required=True` checks that the input to `Schema.load()` provides the field of concern. The field `fields.Method('get_links')` only uses the `serialize` arguments (`'get_links'`), so that field is a "`dump_only` field" (https://marshmallow.readthedocs.io/en/stable/_modules/marshmallow/fields.html#Method). `dump_only` fields are not checked at load time, so the `require=True` doesn't do anything: that's why we should remove it.

Reply:
Didn't know dump_only where not checked. It makes sense, tho it would be good to have them checked just to make sure we make things right.